### ISSUE TITLE
Gamepad not supported

### ISSUE THREAD

**Issue by safak45x** [hmlendea/geforcenow-electron#22]
Is it a problem for me, or is it due to my application?

**Comment by hmlendea** [comment]
It was working well while testing the app in a previous release.
I don't have a gamepad with me right now to re-test though :(

What gamepad are you using?

**Comment by safak45x** [comment]
https://www.monsternotebook.com.tr/pusatgamepad/monster-pusat-pro-gaming-gamepad/

**Comment by safak45x** [comment]
Buddy i have to say
The work you do is seriously revolutionary for linux
Maybe not many people know because it's new now
But as the number of users increases, you will understand what I mean.

**Comment by bachulator** [comment]
Having same problem. Generic Dinput not working. The gamepad seems to be working in other apps normally.

**Comment by ggtylerr** [comment]
Same problem here. Using a switch pro controller, but with Steam on the background with a basic gamepad setting for desktop. Seems to work fine on [Gamepad Tester](https://gamepad-tester.com/) and the web version of GeForce NOW.

**Comment by hmlendea** [comment]
So far I haven't got a solution for this

**Comment by PeterTucker** [comment]
Bump. XBox One S controller does not work.

**Comment by N3rdB01-1** [comment]
Unable to use GameSir Pro controller either. I notice it does work if I use the extension for Chrome browser to play GeForce Now. Possibly may look at that extension to see what they are using to call out the sub routines?

**Comment by seitbekir** [comment]
So, As I used an App, this fully support gamepad. You only need to run Steam, for example. It provides the driver you need.

**Comment by Angel-Tube** [comment]
@hmlendea what controller are you using? doesnt work for me. i use some kind of old logitech controller.

**Comment by hmlendea** [comment]
XBox One Controller (Wireless)

**Comment by Angel-Tube** [comment]
Perhaps I should inquire how you set it up. Is it practically 'plug and play' even though it is wireless?
Should we be seeing some sort of popup?

Mine is wired by USB, btw.

**Comment by hmlendea** [comment]
Hi so no popup on my end or anything like that. It just works.

I will soon get a generic chinese bluetooth gamepad and will try to see how that one behaves.
I will also try with an old cable NES clone gamepad but I'm not sure if that is even a good test at all since it's a very niche gamepad and not really intended for this use case.

**Comment by Jollaholisti** [comment]
Controllers doesn't work on this "app". Seems that this "app" is using some poorly supporting browser version behind it that doesn't support controllers. I hope this app gets at least Google Chrome/Chromium versions running under hood in near future. Now it's much better to use only Google Chromium to play GFN. - Linux User

**Comment by ggtylerr** [comment]
@Jollaholisti Yes, everyone here knows that. That's why this is using Electron, a framework meant to have desktop apps using a web browser. There are plenty of other apps that you might use that also uses it (or its mobile equivalent, React Native.) It's not the best solution (far from it), but it's the only method of bringing GFN onto a third-party app (without reverse-engineering their software.) If you want to use chrome/browsers instead of this app, that's fine. This is just here for anyone that does want to use an app.

This app already uses chromium - and I'm assuming it might just be an outdated version of Electron. But I rarely develop with it.

(Also, no need to mention you're a linux user. Everyone has to use it in order for the app to function.)

**Comment by ripper81** [comment]
Same here controller is not detected. However it is not working with chromium either.

**Comment by Jollaholisti** [comment]
I have no issues to use controllers on Chromium. All other not pure Google
browsers got only that kind of issues.

ti 11. toukok. 2021 klo 18.52 ripper81 ***@***.***) kirjoitti:

**Comment by sutterkaine75** [comment]
Appimagine works on the gamepad 360 so snap don't.

**Comment by predr** [comment]
Regarding snap, you need to add `joystick` plug. This is required for snap app to see controllers. [Publisher in the Snap Store](https://snapcraft.io/geforcenow) can request an auto-connect, otherwise users must connect it manually.

```bash
snap connect geforcenow:joystick
error: snap "geforcenow" has no plug named "joystick"
```

I think something like this:
```yaml
"build": {
  ...
  "snap": {
    "plugs": [
      "default",
      "joystick"
    ]
  },
  ...
}
```

https://www.electron.build/configuration/snap#SnapOptions-plugs
https://snapcraft.io/docs/joystick-interface

**Comment by N3rdB01-1** [comment]
It only seems to work using the flatpak if I have steam. The other issue I am running into is the full screen seems to be borked.

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Comment by hmlendea** [comment]
@predr update comming soon with the fix you mentioned
@N3rdB01-1 could you please detail the fullscreen issues in another issue? An existing one if it fits there or a new one otherwise

**Comment by hmlendea** [comment]
Version 1.1.4 released with #65. Closing this issue for now.
Please feel free to reopen in case the gamepad issues still persist.
